### PR TITLE
Wip osx freebsd

### PR DESCRIPTION
**Pull request by ukernel**
Add error code translation for FreeBSD/OSX. Also fix a few compilation errors.

### CODE REVIEW

File: src/common/Makefile.am [hunk 1]
```diff
@@ -94,12 +89,23 @@ libcommon_internal_la_SOURCES += \
         common/aix_errno.cc
 endif
 
-if WITH_RBD
```
Comment (gregsfortytwo) [inline src/common/Makefile.am]: Why switch it to building regardless of RBD?

**Comment by gregsfortytwo** [comment]
Do these errno files differ from each other at all?

File: src/common/Makefile.am (hunk 1 above)
Comment (ukernel) [inline src/common/Makefile.am]: common/blkdev.cc is used by FileJournal.cc, nothing do with rbd. this line was introduced by solaris port RP, it should be wrong.

**Comment by ukernel** [comment]
On Thu, Dec 3, 2015 at 6:51 AM, Gregory Farnum <notifications@github.com>
wrote:

> Do these errno files differ from each other at all?
>

about four error codes differ

**Comment by trociny** [comment]
@ukernel why did you close this? Do you have plans to submit another PR?

**Comment by ukernel** [comment]
On Mon, Jan 4, 2016 at 8:18 PM, Mykola Golub <notifications@github.com>
wrote:

> @ukernel <https://github.com/ukernel> why did you close this? Do you have
> plans to submit another PR?
>
> —
>
I will open new one

**Pull request closed without merging**